c-scape/src/threads/key.rs

```diff
@@ -0,0 +1,74 @@
+use core::cell::Cell;
+use core::ptr::{null, null_mut};
+use errno::{set_errno, Errno};
+use libc::{c_int, c_void};
+
+const PTHREAD_KEYS_MAX: u32 = 128;
+
+#[thread_local]
+static CURRENT_KEY: Cell<libc::pthread_key_t> = Cell::new(0);
```

Comment:
`CURRENT_KEY` can't be thread-local, because keys are shared between threads. And below, `pthread_key_delete` needs to free the key on all threads. I think what we need is an `RwLock` guarding a shared `CURRENT_KEY` and a shared destructors array.

Reply:
I used a `Mutex` here because I misread, but I don't think it really matters.